neo4j-to-neptune/src/main/java/com/amazonaws/services/neptune/util/GzipCompressUtils.java

```
@@ -0,0 +1,130 @@
+/*
+Copyright 2025 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+Licensed under the Apache License, Version 2.0 (the "License").
+You may not use this file except in compliance with the License.
+A copy of the License is located at
+    http://www.apache.org/licenses/LICENSE-2.0
+or in the "license" file accompanying this file. This file is distributed
+on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+express or implied. See the License for the specific language governing
+permissions and limitations under the License.
+*/
+
+package com.amazonaws.services.neptune.util;
+
+import java.io.BufferedInputStream;
+import java.io.BufferedOutputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collectors;
+import java.util.zip.GZIPOutputStream;
+
+/**
+ * Utility class for gzip compression of CSV files
+ */
+public class GzipCompressUtils {
+    /**
+     * Upload Neptune vertices and edges CSV files asynchronously
+     */
+    public static void compressCsvFiles(String uploadDirectoryPath, Boolean isCompressDelete) throws Exception {
```

Exception is not thrown?

---

The method can still throw a `IllegalStateException` from the directory checking. 


The only exception it swallows, but produces an error message is the compress one. I want this to swallow because i dont want the whole workflow to explode if one file couldnt be compressed.

---

Refer to comment below... But I will need to throw the compress exception as well with the new implementation of optional retaining of original files.